### ISSUE TITLE
EOS apk problems in multiplayer. Host migration issues?

### ISSUE THREAD

**Issue by alefriant** [oculus-samples/Unreal-SharedSpaces#91]
Hi @jasonmeisel 
If you could help me fixing this will be great please. Seems the problem is related to the host migration and I don't have the knowledge to fix that. 

The file was tested without any modification than adding the MetaID and EpicID, etc. No blueprint or Cpp was touched. No automatic modifications by the Meta plugin, nothing. Was tested using the UE 5.4.3 v100 SDK v68 without external plugins than the ones inside the Oculus_Unreal file and inside the Sharedspaces_EOS. 

Notes:
1. Tested in single player everything seems to be fine. 
- Except the transition map, it never appeared.

Issues:
2. I did several tests in multiplayer using the EOS release. (If you need logs I can upload those, but I think with this info is enough)

- [ ] Here the transition map appears only sometimes. (will be described below)
- [ ] When the host is in the Lobby, the client can travel between rooms and come back to the lobby only if the host stay in the lobby, but appears randomly in the exit of the rooms, not in the corresponding room (BP of room exit).

- [x] Case 1: If the host join any room and the client follow the host it works.
- [ ]  Case 1.1: If the host leave, the client appears in the transition map, and stay stuck there forever.
- [x]  Case 1.2: If the client leave, the host can follow the client to the lobby, even if the lobby host wasn't the host of that room.
            
- [x] Case 2: If the client join any room and the client follow the host it works. 
- [ ] Case 2.1: If the host leave, the client appears in the transition map, and stay stuck there forever.
- [x] Case 2.2: If the client leave, the host can follow the client to the lobby, even if the lobby host wasn't the host of that room.

- [ ]  Sometimes the star appears in the rooms, sometimes when the host leave the star takes a long to appear, sometimes never appear.
- [x] Invitations works, using the Meta button and using the Sharedspaces pedestal, and the Meta menu appears correctly, works better than in the Photon version and never showed us the OVR10003 error like in the Photon edition which still has the problem sometimes.
- [x] Two or more hosts can join the purple room from different lobbies and come back to their corresponding lobbies.
- [ ] The green room doesn't work at all, but sometimes when the host is in the lobby and the client goes to the green door, the client will appear after a few seconds in the exit of another room, back in the lobby. 

If you need me to test something in particular in multiplayer please let me know. 

Thanks!

Alejandro Friant.

**Comment by alefriant** [comment]
Note: 
The EOS version seems to take more time to load when the host is being assigned, takes a while to appear the star and to show in the log that you took the room, but once working the scenario seems to be more stable than in the Photon version. Even if there is some delay because we are testing from different countries, you can't notice it because the movement is smooth and the pawn never disappeared to appear in a different location like sometimes happen in Photon.
In summary, we are eagerly waiting to migrate to the EOS version so we can use the EOS voice chat, but we really need the host migration feature because our game relies on the core of SharedSpaces, allowing players to travel and play with friends between rooms while maintaining consistency across the game. This way, some friends can travel to one room while others go to a different room in parallel to achieve a common goal :)

**Comment by alefriant** [comment]
Hey @jasonmeisel !
I forget to mention, in this eos branch the project can be run in the viewport as Standalone, Listen server and Client
That's a great feature compared to Photon version because I can test the replication properly in the viewport and avoid several tests in headset which saves a lot of development time. 
Will be great if you can maintain that feature along the host migration :)  

Thanks again!!

**Comment by alefriant** [comment]
@jasonmeisel 
I won't need this for now, until the Unreal team on the other side fixes the Occlusion Culling issue they have in 5.4.3, because isn't compatible with Tone Mapping and UE 5.3.2 is. 
One of my game mechanic relies in the day night cycle and it's critical. 
Anyways will be great to have the functional EOS version with host migration later when that has been solved. :)

Thank you!
